code/game/machinery/kitchen/cooking_machines/_cooker.dm

```diff
@@ -91,8 +103,8 @@
 		for(var/mob/living/M in cooking_obj.contents)
 			M.apply_damage(rand(30,40), BURN, "chest")
 
-	// Not sure why a food item that passed the previous checks would fail to drop, but safety first.
-	if(!user.unEquip(I))
+	// Not sure why a food item that passed the previous checks would fail to drop, but safety first. (Hint: Borg grippers. That is why. ~Mechoid.)
+	if(!user.unEquip(I) && !istype(user,/mob/living/silicon/robot))
```

Comment:
That's for putting food into a machine, and AI's don't have hands or grippers to do that with. Taking food out, however, is a proximity check, so that fixes the referenced one, but I can test it again.